beacon/goclient/genesisgoclient/adapter.go

```diff
@@ -25,7 +28,14 @@ func (a *adapter) GetAttestationData(slot phase0.Slot, committeeIndex phase0.Com
 }
 
 func (a *adapter) GetBeaconNetwork() genesisspectypes.BeaconNetwork {
-	return genesisspectypes.BeaconNetwork(a.GoClient.GetBeaconNetwork())
+	switch a.GoClient.BeaconConfig().GenesisForkVersionVal {
+	case networkconfig.MainnetBeaconConfig.GenesisForkVersionVal:
+		return genesisspectypes.BeaconNetwork(spectypes.MainNetwork)
+	case networkconfig.HoleskyBeaconConfig.GenesisForkVersionVal:
+		return genesisspectypes.BeaconNetwork(spectypes.HoleskyNetwork)
+	default:
+		return genesisspectypes.BeaconNetwork(spectypes.BeaconTestNetwork)
+	}
```

Comment:
This is not enough, we want to actually use the endpoints provided by the beacon node (like the specs endpoint) to get the beacon network values and not just which one to use.

Reply:
This is just an adapter for the genesis code. Fetching beacon network from beacon node is in `beacon/goclient/spec.go`